packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/utils/getListType.ts

```diff
@@ -1,14 +1,30 @@
 import { ListType } from 'roosterjs-editor-types';
 
-function isABulletList(textBeforeCursor: string) {
-    const hasTriggers = ['*', '-', '>'].indexOf(textBeforeCursor[0]) > -1;
-    const REGEX: RegExp = /^(.*?)=>|^(.*?)->|^(.*?)-->|^(.*?)=>|^(.*?)--/;
-    return hasTriggers || REGEX.test(textBeforeCursor);
+function isABulletList(trigger: string) {
+    const triggerLength = trigger.length;
+    if (triggerLength === 1) {
+        const REGEX: RegExp = /^((.*?)-|^(.*?)*|^(.*?)>)$/;
+        return REGEX.test(trigger);
+    } else if (triggerLength === 2) {
+        const REGEX: RegExp = /^((.*?)=>|^(.*?)->|^(.*?)=>|^(.*?)--)$/;
+        return REGEX.test(trigger);
+    } else if (triggerLength === 3) {
+        const REGEX: RegExp = /^(.*?)-->$/;
```

Comment:
same

Reply:
Generally, you can just simply do
const BulletTriggers = ['*', '-', '>', '=>', ... ];
...

if (BulletTriggers.indexOf(trigger) >= 0) { ...

We should try to avoid using Regex when user is typing since regex matching is much slower than just string comparing.

Reply:
Generally, you can just simply do
const BulletTriggers = ['*', '-', '>', '=>', ... ];
...

if (BulletTriggers.indexOf(trigger) >= 0) { ...

We should try to avoid using Regex when user is typing since regex matching is much slower than just string comparing.